src/internal/packager2/layout/create.go

```diff
@@ -0,0 +1,475 @@
+		}
+
+		for valuesIdx, path := range chart.ValuesFiles {
+			if helpers.IsURL(path) {
+				continue
+			}
+
+			rel := fmt.Sprintf("%s-%d", helm.StandardName(string(ValuesComponentDir), chart), valuesIdx)
+			component.Charts[chartIdx].ValuesFiles[valuesIdx] = rel
+
+			if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, path), filepath.Join(compBuildPath, rel)); err != nil {
+				return fmt.Errorf("unable to copy chart values file %s: %w", path, err)
+			}
+		}
+	}
+
+	for filesIdx, file := range component.Files {
+		if helpers.IsURL(file.Source) {
+			continue
+		}
+
+		rel := filepath.Join(string(FilesComponentDir), strconv.Itoa(filesIdx), filepath.Base(file.Target))
+		dst := filepath.Join(compBuildPath, rel)
+		destinationDir := filepath.Dir(dst)
+
+		if file.ExtractPath != "" {
+			if err := archiver.Extract(filepath.Join(packagePath, file.Source), file.ExtractPath, destinationDir); err != nil {
+				return fmt.Errorf(lang.ErrFileExtract, file.ExtractPath, file.Source, err.Error())
+			}
+
+			// Make sure dst reflects the actual file or directory.
+			updatedExtractedFileOrDir := filepath.Join(destinationDir, file.ExtractPath)
+			if updatedExtractedFileOrDir != dst {
+				if err := os.Rename(updatedExtractedFileOrDir, dst); err != nil {
+					return fmt.Errorf(lang.ErrWritingFile, dst, err)
+				}
+			}
+		} else {
+			if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, file.Source), dst); err != nil {
+				return fmt.Errorf("unable to copy file %s: %w", file.Source, err)
+			}
+		}
+
+		// Change the source to the new relative source directory (any remote files will have been skipped above)
+		component.Files[filesIdx].Source = rel
+
+		// Remove the extractPath from a skeleton since it will already extract it
+		component.Files[filesIdx].ExtractPath = ""
+
+		// Abort packaging on invalid shasum (if one is specified).
+		if file.Shasum != "" {
+			if err := helpers.SHAsMatch(dst, file.Shasum); err != nil {
+				return err
+			}
+		}
+
+		if file.Executable || helpers.IsDir(dst) {
+			err = os.Chmod(dst, helpers.ReadWriteExecuteUser)
+			if err != nil {
+				return err
+			}
+		} else {
+			err = os.Chmod(dst, helpers.ReadWriteUser)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for dataIdx, data := range component.DataInjections {
+		rel := filepath.Join(string(DataComponentDir), strconv.Itoa(dataIdx), filepath.Base(data.Target.Path))
+		dst := filepath.Join(compBuildPath, rel)
+
+		if err := helpers.CreatePathAndCopy(filepath.Join(packagePath, data.Source), dst); err != nil {
+			return fmt.Errorf("unable to copy data injection %s: %s", data.Source, err.Error())
+		}
+
+		component.DataInjections[dataIdx].Source = rel
+	}
+
+	// Iterate over all manifests.
+	for manifestIdx, manifest := range component.Manifests {
+		for fileIdx, path := range manifest.Files {
+			rel := filepath.Join(string(ManifestsComponentDir), fmt.Sprintf("%s-%d.yaml", manifest.Name, fileIdx))
```

Comment:
This is an issue with the current skeleton, but packages that have remote manifests will break here. I believe a `helpers.IsUrl()` check will solve this

Reply:
If that is a current issue with Skeleton packages I think it is fine if the new implementation also has it. We need to set the DoD somewhere.

Reply:
Fair enough, I'll make an issue

Reply:
#3082